### PR TITLE
[BEAM-13899] Improve coverage of debug package

### PR DESCRIPTION
**Pull request by damccorm**
Improves testing of the debug package. Mostly sets up pipelines and verifies the results (with testing print I had to do some log capturing).

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Review by jrmccluskey — APPROVED** [review]
LGTM!

**Comment by jrmccluskey** [comment]
R: @lostluck @youngoli

**Review by lostluck — CHANGES_REQUESTED** [review]
Same quick ptest changes here.

I do like the capturing of the logger output.

debug.Print is bare bones, but it doesn't have to be elaborate.

File: sdks/go/pkg/beam/x/debug/head_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debug
+
+import (
+	"testing"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	_ "github.com/apache/beam/sdks/v2/go/pkg/beam/runners/direct"
```
Comment (lostluck) [inline sdks/go/pkg/beam/x/debug/head_test.go:22]: Just removing the dead imports.
```suggestion
```

Comment (lostluck) [inline sdks/go/pkg/beam/x/debug/head_test.go]:
```suggestion
```

File: sdks/go/pkg/beam/x/debug/print_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+import (
+	"bytes"
+	"log"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	_ "github.com/apache/beam/sdks/v2/go/pkg/beam/runners/direct"
```
Comment (lostluck) [inline sdks/go/pkg/beam/x/debug/print_test.go:26]:
```suggestion
```

**Review by lostluck — APPROVED** [review]
LGTM, just need to wait for the tests to pass after my import removals.

**Pull request merged**